internal-dns/types/versions/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Versioned types for the DNS server API.
+//!
+//! # Organization
+//!
+//! Types are organized based on the rules outlined in [RFD
+//! 619](https://rfd.shared.oxide.computer/rfd/0619).
+
+pub mod latest;
+#[path = "initial/mod.rs"]
+pub mod v1;
+#[path = "soa_and_ns/mod.rs"]
+pub mod v2;
```

Comment:
This idea is very similar to how database migrations use named directories + a canonical ordered list. It should result in no merge conflicts in implementations (which are terrible to deal with) but instead, merge conflicts here and possibly in `latest.rs` which are much easier to manage.